src/ansi.c

```diff
@@ -85,92 +85,81 @@ draw_ansi(struct view *view, int *ansi_num, char **ansi_ptrs, int max_width, siz
-			if (strcmp(ansi_code, "3") == 0)
+			if (strcmp(ansi_code_part, "3") == 0)
 				cur_ansi_status.attr = A_ITALIC;
-			if (strcmp(ansi_code, "4") == 0)
+			if (strcmp(ansi_code_part, "4") == 0)
 				cur_ansi_status.attr = A_UNDERLINE;
-			if (strcmp(ansi_code, "5") == 0)
+			if (strcmp(ansi_code_part, "5") == 0)
 				cur_ansi_status.attr = A_BLINK;
-			if (strcmp(ansi_code, "6") == 0)
+			if (strcmp(ansi_code_part, "6") == 0)
 				cur_ansi_status.attr = A_BLINK; // This is supposed to be faster than normal blink, but ncurses doesn't have any way to achieve.
-			if (strcmp(ansi_code, "7") == 0)
+			if (strcmp(ansi_code_part, "7") == 0)
 				cur_ansi_status.attr = A_REVERSE;
-			if (strcmp(ansi_code, "8") == 0)
+			if (strcmp(ansi_code_part, "8") == 0)
 				cur_ansi_status.attr = A_INVIS;
-			if (strcmp(ansi_code, "9") == 0)
+			if (strcmp(ansi_code_part, "9") == 0)
 				// This is supposed to be strikethrough, but ncurses doesn't have any way to achieve.
-			if (strcmp(ansi_code, "30") == 0)
+			if (strcmp(ansi_code_part, "30") == 0)
 				cur_ansi_status.fg = COLOR_BLACK;
-			if (strcmp(ansi_code, "31") == 0)
+			if (strcmp(ansi_code_part, "31") == 0)
 				cur_ansi_status.fg = COLOR_RED;
-			if (strcmp(ansi_code, "32") == 0)
+			if (strcmp(ansi_code_part, "32") == 0)
 				cur_ansi_status.fg = COLOR_GREEN;
-			if (strcmp(ansi_code, "33") == 0)
+			if (strcmp(ansi_code_part, "33") == 0)
 				cur_ansi_status.fg = COLOR_YELLOW;
-			if (strcmp(ansi_code, "34") == 0)
+			if (strcmp(ansi_code_part, "34") == 0)
 				cur_ansi_status.fg = COLOR_BLUE;
-			if (strcmp(ansi_code, "35") == 0)
+			if (strcmp(ansi_code_part, "35") == 0)
 				cur_ansi_status.fg = COLOR_MAGENTA;
-			if (strcmp(ansi_code, "36") == 0)
+			if (strcmp(ansi_code_part, "36") == 0)
 				cur_ansi_status.fg = COLOR_CYAN;
-			if (strcmp(ansi_code, "37") == 0)
+			if (strcmp(ansi_code_part, "37") == 0)
 				cur_ansi_status.fg = COLOR_WHITE;
-			if (strcmp(ansi_code, "40") == 0)
+			if (strcmp(ansi_code_part, "38") == 0) {
+				short c256 = convert_ansi_into_256_color(saveptr);
+				if (c256 != -1)
+					cur_ansi_status.fg = c256;
+			}
+			if (strcmp(ansi_code_part, "40") == 0)
 				cur_ansi_status.bg = COLOR_BLACK;
-			if (strcmp(ansi_code, "41") == 0)
+			if (strcmp(ansi_code_part, "41") == 0)
 				cur_ansi_status.bg = COLOR_RED;
-			if (strcmp(ansi_code, "42") == 0)
+			if (strcmp(ansi_code_part, "42") == 0)
 				cur_ansi_status.bg = COLOR_GREEN;
-			if (strcmp(ansi_code, "43") == 0)
+			if (strcmp(ansi_code_part, "43") == 0)
 				cur_ansi_status.bg = COLOR_YELLOW;
-			if (strcmp(ansi_code, "44") == 0)
+			if (strcmp(ansi_code_part, "44") == 0)
 				cur_ansi_status.bg = COLOR_BLUE;
-			if (strcmp(ansi_code, "45") == 0)
+			if (strcmp(ansi_code_part, "45") == 0)
 				cur_ansi_status.bg = COLOR_MAGENTA;
-			if (strcmp(ansi_code, "46") == 0)
+			if (strcmp(ansi_code_part, "46") == 0)
 				cur_ansi_status.bg = COLOR_CYAN;
-			if (strcmp(ansi_code, "47") == 0)
```

Comment:
Is it intentional to delete the "47" code?

Reply:
Sorry, it's just my mistake. 
Thanks for catching, Fixed here b86f198 .